dali/pipeline/util/thread_pool.cc

```diff
@@ -150,12 +151,8 @@ void ThreadPool::ThreadMain(int thread_id, int device_id, bool set_affinity,
       nvml::SetCPUAffinity(core);
     }
 #endif
-  } catch (std::exception &e) {
-    std::lock_guard lock(error_mutex_);
```

Comment:
The errors are per-thread and are not accessed until the runners are idle - no need for a lock.